libzstd/src/lib.rs

```diff
@@ -29,7 +29,21 @@ pub unsafe extern "C" fn compress_scroll_batch_bytes(
     let src = unsafe { slice::from_raw_parts(src, src_size as usize) };
     let out = unsafe { slice::from_raw_parts_mut(output_buf, buf_size as usize) };
 
-    let mut encoder = init_zstd_encoder(N_BLOCK_SIZE_TARGET);
+    // when the src size larger than one input block (so the output has too
+    // be distributed in multiple blocks), we average each input block to
+    // avoid the small size in last input, which may lead to Raw or RLE blocks
+    let mut encoder = if src_size > N_BLOCK_SIZE_TARGET as u64{
+        let exp_blocks = (src_size - 1) / N_BLOCK_SIZE_TARGET as u64 + 1;
+        /* 
+            slightly increase for blocks except the last one,
+            so ensure no "tip" would be left
+        */
+        let reset_blk_size = src_size / exp_blocks + 8;
+        assert!(reset_blk_size <= N_BLOCK_SIZE_TARGET as u64);
```

Comment:
if `src_size == 2 * N_BLOCK_SIZE_TARGET - 2`, looks like it will break the assertion in this line.

Reply:
It is the case. I think it should be ok to drop this "add small tip" trick.

Reply:
It is the case. I think it should be ok to drop this "add small tip" trick.